### ISSUE TITLE
Core doesn't publish to secondary history archive (e.g. AWS S3) if first archive failed (cache)

### ISSUE THREAD

**Issue by oryband** [stellar/stellar-core#1649]
# Issue type
Bug

# Version found
- 9.1.0
- 9.2.0

# Your environment and setup
- Dockerized Ubuntu 16.04 (https://hub.docker.com/_/ubuntu/)

# Issue description
I've set up two history archive: cache and AWS S3:
```toml
[HISTORY.cache]
get="cp /tmp/stellar-core/history/vs/{0} {1}"
put="cp {0} /tmp/stellar-core/history/vs/{1}"
mkdir="mkdir -p /tmp/stellar-core/history/vs/{0}"

[HISTORY.s3]
get="aws s3 cp s3://my-bucket/{0} {1}"
put="aws s3 cp {0} s3://my-bucket/{1}"
# mkdir isn't necessary for aws cli, 'cp' command takes care of that
```

I've accidentally only initialized (`--newhist`) the S3 archive, and forgot `cache`. Core tried to published to `cache` and failed, and didn't try to publish to S3 afterwards, even though it should have.

## Supporting files

Logs where full of the following lines:
```
cp: cannot stat '/tmp/stellar-core/history/vs/history/00/05/99/history-0005993f.json': No such file or directory
GBT6G [Process WARNING] process 1829 exited 1: cp /tmp/stellar-core/history/vs/history/00/05/99/history-0005993f.json buckets/tmp/history-a05d40f6bb31c070/stellar-history.json
GBT6G [Work WARNING] Scheduling retry #1/5 in 2 sec, for get-remote-file history/00/05/99/history-0005993f.json
```
However, there's was nothing else in the logs indicating publishing to S3 was skipped.

**Comment by edwardsun888** [comment]
I met the same issue, it maybe No permission to do mkdir="mkdir -p /tmp/stellar-core/history/vs/{0}".
but I do not know to solve this problem currently.

**Comment by benxage** [comment]
@edwardsun888 did you find a solution to the problem? It seems unnecessary for there to be an `mkdir` command since s3 automatically creates those folders for you.

**Comment by oryband** [comment]
> I met the same issue, it maybe No permission to do mkdir="mkdir -p /tmp/stellar-core/history/vs/{0}".
> but I do not know to solve this problem currently.

The topic of this issue is not the error that I accidently made, but the fact the Core doesn't publish to secondary history archive (e.g. AWS S3) if first archive failed for whatever reason.

> did you find a solution to the problem? It seems unnecessary for there to be an `mkdir` command since s3 automatically creates those folders for you.

You're correct - there's no reason to use `mkdir` for publishing to S3.

**Comment by marta-lokhova** [comment]
Resolved by #2063
